modules/fedimint-wallet-client/src/lib.rs

```diff
@@ -453,89 +539,176 @@ impl WalletClientModule {
             })?;
 
-        Ok((operation_id, address))
+        Ok((operation_id, address, tweak_idx))
     }
 
-    pub async fn subscribe_deposit_updates(
+    pub async fn find_tweak_idx_by_operation_id(
         &self,
         operation_id: OperationId,
-    ) -> anyhow::Result<UpdateStreamOrOutcome<DepositState>> {
-        let operation_log_entry = self
+    ) -> anyhow::Result<TweakIdx> {
+        Ok(self
             .client_ctx
-            .get_operation(operation_id)
+            .module_db()
+            .clone()
+            .begin_transaction_nc()
             .await
-            .with_context(|| anyhow!("Operation not found: {}", operation_id.fmt_short()))?;
+            .find_by_prefix(&PegInTweakIndexPrefix)
+            .await
+            .filter(|(_k, v)| future::ready(v.operation_id == operation_id))
+            .next()
+            .await
+            .ok_or_else(|| anyhow::format_err!("OperationId not found"))?
+            .0
+             .0)
+    }
 
-        if operation_log_entry.operation_module_kind() != WalletCommonInit::KIND.as_str() {
-            bail!("Operation is not a wallet operation");
+    pub async fn get_pegin_tweak_idx(
+        &self,
+        tweak_idx: TweakIdx,
+    ) -> anyhow::Result<PegInTweakIndexData> {
+        self.client_ctx
+            .module_db()
+            .clone()
+            .begin_transaction_nc()
+            .await
+            .get_value(&PegInTweakIndexKey(tweak_idx))
+            .await
+            .ok_or_else(|| anyhow::format_err!("TweakIdx not found"))
+    }
+
+    pub async fn get_claimed_pegins(
+        &self,
+        dbtx: &mut DatabaseTransaction<'_>,
+        tweak_idx: TweakIdx,
+    ) -> Vec<(
+        bitcoin::OutPoint,
+        TransactionId,
+        Vec<fedimint_core::OutPoint>,
+    )> {
+        let outpoints = dbtx
+            .get_value(&PegInTweakIndexKey(tweak_idx))
+            .await
+            .map(|v| v.claimed)
+            .unwrap_or_default();
+
+        let mut res = vec![];
+
+        for outpoint in outpoints {
+            let claimed_peg_in_data = dbtx
+                .get_value(&ClaimedPegInKey {
+                    peg_in_index: tweak_idx,
+                    btc_out_point: outpoint,
+                })
+                .await
+                .expect("Must have a corresponding claim record");
+            res.push((
+                outpoint,
+                claimed_peg_in_data.claim_txid,
+                claimed_peg_in_data.change,
+            ));
         }
 
-        let operation_meta = operation_log_entry.meta::<WalletOperationMeta>();
+        res
+    }
 
-        if !matches!(
-            operation_meta.variant,
-            WalletOperationMetaVariant::Deposit { .. }
-        ) {
-            bail!("Operation is not a deposit operation");
-        }
+    /// Like [`Self::recheck_pegin_address`] but by `operation_id`
+    pub async fn recheck_pegin_address_by_op_id(
```

Comment:
I'm a bit conflited on the "what should we key the peg-in checks on".

Before we *had* to use operation_id because checking was a state_machine. So to keep the CLI APIs as they were I needed functions that work on operatin_id. Now it's just an index of derived address (`TweakIdx`) and it seems easier. But we could also look up things by `address` to avoid users need to think about any indexes...

Anyway, I think it's OK to keep, as it's public and can be used by external apps. I don't think it hurts much.

Reply:
If you key on address won't that fail reuse? And if someone does reuse a pegin address can't anyone watching just publish the inclusion proof + the info from the previous pegin to that address in order to receive the ecash?

Reply:
There's one address for one operation_id with one tweak_idx. address and operation_id are both derived from the tweak_idx. Each of these can have multiple deposits though.

Reply:
> And if someone does reuse a pegin address can't anyone watching just publish the inclusion proof + the info from the previous pegin to that address in order to receive the ecash?

I believe that's correct, if you know the tweak key and the deposit has enough confs you can construct a valid wallet input and claim the ecash.

Reply:
Scratch that, it's not possible. An observer will know the public key used to tweak the peg-in descriptor, however they would also need to know the secret, which is used to sign the fedimint tx that claims the input.

https://github.com/fedimint/fedimint/blob/ef6d4b12241651c2faa842547cbc368ceabbe4e5/fedimint-client/src/transaction/builder.rs#L123